.github/workflows/ci.yml

```diff
@@ -55,72 +52,3 @@ jobs:
       # Check minimal build. `tests/seek.rs` fails if there are no decoders at all,
       # adding one to make the tests check pass.
       - run: cargo check --tests --lib --no-default-features --features mp3
-  cargo-publish:
-    if: github.event_name == 'push' && github.ref == 'refs/heads/master'
-    env:
-      CRATESIO_TOKEN: ${{ secrets.CRATESIO_TOKEN }}
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v4
-      - name: Update apt
-        run: sudo apt update
-      - name: Install alsa
-        run: sudo apt install -y --no-install-recommends libasound2-dev pkg-config
-      - name: Run cargo publish for rodio
-        continue-on-error: true
-        run: |
-          RODIO_TMP=$(mktemp /tmp/rodioXXX.txt) || echo "::error::mktemp error"
-          echo "RODIO_TMP=$RODIO_TMP" >> $GITHUB_ENV
-          cargo publish --token $CRATESIO_TOKEN 2> $RODIO_TMP
-      - name: Check if rodio is already published
-        run: |
-          empty=0
-          RODIO_TMP="${{ env.RODIO_TMP }}"
-          grep -q '[^[:space:]]' < $RODIO_TMP || empty=1
-          [ $empty -eq 0 ] && cat $RODIO_TMP
-          [ $empty -eq 1 ] || grep -q "is already uploaded" < $RODIO_TMP
-
-  create-git-tag:
-    if: github.event_name == 'push' && github.ref == 'refs/heads/master'
-    runs-on: ubuntu-latest
-    steps:
-    - name: Check if WORKFLOW_TOKEN is set
-      run: |
-        if [ -z "${{ secrets.WORKFLOW_TOKEN }}" ]; then
-          echo "Personal access token WORKFLOW_TOKEN is not set"
-          exit 1
-        else
-          echo "Checked `WORKFLOW_TOKEN` is set"
-        fi
-    - name: Checkout code
-      uses: actions/checkout@v4
-      with:
-        fetch-depth: 0  # Fetch all history to list all existing tags
-        token: ${{ secrets.WORKFLOW_TOKEN }}
-    - name: Extract version from Cargo.toml
-      id: extract_version
-      run: |
-        version=$(awk '/\[package\]/,/^version/ { if ($1 == "version") { gsub(/"/, "", $3); print $3 } }' Cargo.toml)
-        echo "Version value found: $version"
-        echo "version=$version" >> $GITHUB_OUTPUT
-    - name: Check if tag exists
-      id: check_tag
-      run: |
-        version=${{ steps.extract_version.outputs.version }}
-        version_name="v$version"
-        if git rev-parse "refs/tags/$version_name" >/dev/null 2>&1; then
-          echo "Tag $version_name already exists"
-          echo "tag_exists=true" >> $GITHUB_OUTPUT
-        else
-          echo "Tag $version_name does not exist"
-          echo "tag_exists=false" >> $GITHUB_OUTPUT
-        fi
-    - name: Create and push tag
```

Comment:
Splitting these steps into separate runs, in turn required to pass things between them, which is too complex. 
Also sh/bash convention is to use uppercase for variable names.
(If really necessary .inputs and .outputs seems to be [the way to do it](https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#example-defining-outputs-for-a-job)).

Reply:
it is such pain to pass things between steps..... Honestly for my other projects the github CI just calls a bash file. Bash is just as cursed as these workflows but at least I know bash and I can test it locally.